sos/report/plugins/ceph_osd.py

```diff
@@ -0,0 +1,100 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+import glob
+
+
+class CephOSD(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    short_desc = 'CEPH osd'
+
+    plugin_name = 'ceph_osd'
+    profiles = ('storage', 'virt', 'container')
+    containers = ('ceph-osd.*',)
+
+    def check_enabled(self):
+        return True if glob.glob('/var/lib/ceph/osd/*/*') else False
+
+    def setup(self):
+        self.add_file_tags({
+            '/var/log/ceph/ceph-osd.*.log': 'ceph_osd_log',
+        })
+
+        # Only collect OSD specific files
+        self.add_copy_spec([
+            "/var/log/ceph/ceph-osd*.log",
+            "/var/log/ceph/ceph-volume*.log",
+
+            "/var/lib/ceph/osd/",
+            "/var/lib/ceph/bootstrap-osd/",
+
+            "/run/ceph/ceph-osd*"
+        ])
+
+        self.add_cmd_output([
+            "ceph osd metadata",
+            "ceph osd erasure-code-profile ls",
+            "ceph osd crush show-tunables",
+            "ceph osd crush dump"
+        ])
+
+        ceph_cmds = [
+            "osd tree",
+            "osd stat",
+            "osd df tree",
+            "osd dump",
+            "osd df",
+            "osd perf",
+            "osd blocked-by",
+            "osd pool ls detail",
+            "osd pool autoscale-status",
+            "osd numa-status"
+        ]
+
+        # is this not to be collected when no containers?
```

Comment:
I suspect this was an oversight. I hink you can add this to `ceph_cmds` instead.

Reply:
Understood. But then except mon, everything else mostly becomes an empty stub for now and the ceph_common one just would collect just ceph -v for now, and logs like  

            "/var/log/ceph/ceph.audit.log*",
            "/var/log/calamari/*.log",
            "/etc/ceph/",
            "/etc/calamari/",
            "/var/lib/ceph/tmp/",

Does that sound ok?

I also think the OSD plugin could collect command output like ceph-disk list and ceph-volume inventory? Those would run without needing the admin keys. If you don't want to add any new ones though, osd would become a stub as well since there's nothing that wouldn't run on mon (but would on OSD) in the current list.

Reply:
> [...]
> Does that sound ok?

Yes, that's OK. OSD list wouldn't be empty - at least, `ceph-volume lvm list` and `ceph-disk list` commands can be run OSD.

Reply:
ack